src/server/node_manager.c

```diff
@@ -1143,13 +1290,26 @@ set_vnode_state(struct pbsnode *pnode, unsigned long state_bits, enum vnode_stat
 		pnode->nd_attr[(int)ND_ATR_state].at_flags |= ATR_SET_MOD_MCACHE;
 	}
 
-	if (nd_prev_state != pnode->nd_state) {
+	/* If the state has changed update the last change time */
+	if (vnode_o->nd_state != pnode->nd_state) {
 		char str_val[STR_TIME_SZ];
-
 		snprintf(str_val, sizeof(str_val), "%d", time_int_val);
 		set_attr_generic(&(pnode->nd_attr[(int)ND_ATR_last_state_change_time]),
-			&node_attr_def[(int) ND_ATR_last_state_change_time], str_val, NULL, SET);
-	}
+			&node_attr_def[(int) ND_ATR_last_state_change_time], str_val, NULL,
+			SET);
+	}
+
+	/* Write the vnode state change event to server log */
+	last_time_int = (int)vnode_o->nd_attr[(int)ND_ATR_last_state_change_time]
+		.at_val.at_long;
+	snprintf(local_log_buffer, LOG_BUF_SIZE-1,
+		"set_vnode_state;vnode.state=0x%lx vnode_o.state=0x%lx "
+		"vnode.last_state_change_time=%d vnode_o.last_state_change_time=%d "
+		"state_bits=0x%lx state_bit_op_type_str=%s state_bit_op_type_enum=%d",
+		pnode->nd_state, vnode_o->nd_state, time_int_val, last_time_int,
+		state_bits, get_vnode_state_op(type), type);
+	log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_NODE, LOG_INFO,
```

Comment:
Please use log_eventf() so the need for local_log_buffer can be taken out.

Reply:
Done